nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/PutTCP.java

```diff
@@ -0,0 +1,270 @@
+     *
+     * @param context
+     *            - the current process context.
+     *
+     * @return The URI value as a String.
+     */
+    @Override
+    protected String createTransitUri(final ProcessContext context) {
+        final String protocol = TCP_VALUE.getValue();
+        final String host = context.getProperty(HOSTNAME).getValue();
+        final String port = context.getProperty(PORT).getValue();
+
+        return new StringBuilder().append(protocol).append("://").append(host).append(":").append(port).toString();
+    }
+
+    /**
+     * Get the additional properties that are used by this processor.
+     *
+     * @return List of PropertyDescriptors describing the additional properties.
+     */
+    @Override
+    protected List<PropertyDescriptor> getAdditionalProperties() {
+        return Arrays.asList(CONNECTION_PER_FLOWFILE, OUTGOING_MESSAGE_DELIMITER, TIMEOUT);
+    }
+
+    /**
+     * event handler method to handle the FlowFile being forwarded to the Processor by the framework. The FlowFile contents is sent out over a TCP connection using an acquired ChannelSender object. If
+     * the FlowFile contents was sent out successfully then the FlowFile is forwarded to the success relationship. If an error occurred then the FlowFile is forwarded to the failure relationship.
+     *
+     * @param context
+     *            - the current process context.
+     *
+     * @param sessionFactory
+     *            - a factory object to obtain a process session.
+     */
+    @Override
+    public void onTrigger(final ProcessContext context, final ProcessSessionFactory sessionFactory) throws ProcessException {
+        final ProcessSession session = sessionFactory.createSession();
+        final FlowFile flowFile = session.get();
+        if (flowFile == null) {
+            pruneIdleSenders(context.getProperty(IDLE_EXPIRATION).asTimePeriod(TimeUnit.MILLISECONDS).longValue());
+            context.yield();
+            return;
+        }
+
+        ChannelSender sender = acquireSender(context, session, flowFile);
+        if (sender == null) {
+            return;
+        }
+
+        try {
+            String delimiter = getDelimiter(context, flowFile);
+            ByteArrayOutputStream content = readContent(session, flowFile);
+            if (delimiter != null) {
+                content = appendDelimiter(content, delimiter);
+            }
+            sender.send(content.toByteArray());
+            session.transfer(flowFile, REL_SUCCESS);
+            session.commit();
+        } catch (Exception e) {
+            getLogger().error("Exception while handling a process session, transferring {} to failure.", new Object[] { flowFile }, e);
+            onFailure(context, session, flowFile);
+        } finally {
+            // If we are going to use this sender again, then relinquish it back to the pool.
+            if (!isConnectionPerFlowFile(context)) {
```

Comment:
depending on my previous comment this would potentially change to just calling relinquishSender